allennlp/nn/util.py

```diff
@@ -392,3 +392,10 @@ def inner_device_mapping(storage: torch.Storage, location) -> torch.Storage:  #
         else:
             return storage
     return inner_device_mapping
+
+def ones_like(tensor: torch.Tensor) -> torch.Tensor:
+    """
+    Use clone() + copy_() to make sure that a tensor ends up on the right
+    device at runtime.
+    """
+    return tensor.clone().copy_(torch.ones(tensor.shape))
```

Comment:
I think it'd be more efficient to just use `.fill_(1)` instead of the `.copy()` here.